### ISSUE TITLE
Error in rendering terminal icon in dev auth screen.

### ISSUE THREAD

**Issue by jainkuniya** [zulip/zulip-mobile#2209]
<img width="285" alt="screen shot 2018-04-03 at 7 09 07 pm" src="https://user-images.githubusercontent.com/18511177/38252756-9f57307c-3772-11e8-88b0-e641f879367e.png">

submitted https://github.com/oblador/react-native-vector-icons/pull/693

```
ExceptionsManager.js:73 Warning: Failed prop type: Invalid prop `name` of value `terminal` supplied to `Icon`, expected one of ["activity","airplay","alert-circle","alert-octagon","alert-triangle","align-center","align-justify","align-left","align-right","anchor","aperture","arrow-down","arrow-down-left","arrow-down-right","arrow-left","arrow-right","arrow-up","arrow-up-left","arrow-up-right","at-sign","award","bar-chart","bar-chart-2","battery","battery-charging","bell","bell-off","bluetooth","bold","book","bookmark","box","briefcase","calendar","camera","camera-off","cast","check","check-circle","check-square","chevron-down","chevron-left","chevron-right","chevron-up","chevrons-down","chevrons-left","chevrons-right","chevrons-up","chrome","circle","clipboard","clock","cloud","cloud-drizzle","cloud-lightning","cloud-off","cloud-rain","cloud-snow","codepen","command","compass","copy","corner-down-left","corner-down-right","corner-left-down","corner-left-up","corner-right-down","corner-right-up","corner-up-left","corner-up-right","cpu","credit-card","crop","crosshair","delete","disc","download","download-cloud","droplet","edit","edit-2","edit-3","external-link","eye","eye-off","facebook","fast-forward","feather","file","file-minus","file-plus","file-text","film","filter","flag","folder","github","gitlab","globe","grid","hash","headphones","heart","help-circle","home","image","inbox","info","instagram","italic","layers","layout","life-buoy","link","link-2","list","loader","lock","log-in","log-out","mail","map","map-pin","maximize","maximize-2","menu","message-circle","message-square","mic","mic-off","minimize","minimize-2","minus","minus-circle","minus-square","monitor","moon","more-horizontal","more-vertical","move","music","navigation","navigation-2","octagon","package","paperclip","pause","pause-circle","percent","phone","phone-call","phone-forwarded","phone-incoming","phone-missed","phone-off","phone-outgoing","pie-chart","play","play-circle","plus","plus-circle","plus-square","pocket","power","printer","radio","refresh-ccw","refresh-cw","repeat","rewind","rotate-ccw","rotate-cw","save","scissors","search","server","settings","share","share-2","shield","shopping-cart","shuffle","sidebar","skip-back","skip-forward","slack","slash","sliders","smartphone","speaker","square","star","stop-circle","sun","sunrise","sunset","tablet","tag","target","thermometer","thumbs-down","thumbs-up","toggle-left","toggle-right","trash","trash-2","trending-down","trending-up","triangle","tv","twitter","type","umbrella","underline","unlock","upload","upload-cloud","user","user-check","user-minus","user-plus","user-x","users","video","video-off","voicemail","volume","volume-1","volume-2","volume-x","watch","wifi","wifi-off","wind","x","x-circle","x-square","zap","zoom-in","zoom-out"].
    in Icon (at Icons.js:35)
    in IconTerminal (at ZulipButton.js:83)
    in RCTView (at View.js:78)
    in View (at ZulipButton.js:82)
    in RCTView (at View.js:78)
    in View (at Touchable.ios.js:25)
    in RCTView (at View.js:78)
    in View (at TouchableHighlight.js:306)
    in TouchableHighlight (at Touchable.ios.js:19)
    in Unknown (at ZulipButton.js:81)
    in RCTView (at View.js:78)
    in View (at ZulipButton.js:80)
    in ButtonNormal (at ZulipButton.js:129)
    in ZulipButton (at AuthButton.js:24)
    in AuthButton (at AuthScreen.js:150)
    in RCTView (at View.js:78)
    in View (at Centerer.js:41)
    in RCTView (at View.js:78)
    in View (at Centerer.js:40)
    in Centerer (at AuthScreen.js:144)
    in RCTScrollContentView (at ScrollView.js:793)
    in RCTScrollView (at ScrollView.js:889)
    in ScrollView (at Screen.js:76)
    in RCTView (at View.js:78)
    in View (at KeyboardAvoidingView.js:198)
    in KeyboardAvoidingView (at Screen.js:70)
    in RCTView (at View.js:78)
    in View (at Screen.js:67)
    in Screen (created by Connect(Screen))
    in Connect(Screen) (at AuthScreen.js:143)
    in AuthScreen (created by Connect(AuthScreen))
    in Connect(AuthScreen) (at SceneView.js:17)
    in SceneView (at CardStack.js:466)
    in RCTView (at View.js:78)
    in View (at createAnimatedComponent.js:147)
    in AnimatedComponent (at Card.js:12)
    in Card (at PointerEventsContainer.js:39)
    in Container (at CardStack.js:498)
    in RCTView (at View.js:78)
    in View (at CardStack.js:414)
    in RCTView (at View.js:78)
    in View (at CardStack.js:413)
    in CardStack (at CardStackTransitioner.js:67)
    in RCTView (at View.js:78)
    in View (at Transitioner.js:142)
    in Transitioner (at CardStackTransitioner.js:19)
    in CardStackTransitioner (at StackNavigator.js:41)
    in Unknown (at createNavigator.js:13)
    in Navigator (at createNavigationContainer.js:226)
    in NavigationContainer (at AppWithNavigationState.js:39)
    in AppWithNavigation (created by Connect(AppWithNavigation))
    in Connect(AppWithNavigation) (at ZulipMobile.js:26)
    in Dummy (at StylesProvider.js:46)
    in StyleProvider (created by Connect(StyleProvider))
    in Connect(StyleProvider) (at ZulipMobile.js:25)
    in IntlProvider (at TranslationProvider.js:24)
    in TranslationProvider (created by Connect(TranslationProvider))
    in Connect(TranslationProvider) (at ZulipMobile.js:24)
    in AppDataFetcher (created by Connect(AppDataFetcher))
    in Connect(AppDataFetcher) (at ZulipMobile.js:23)
    in RCTView (at View.js:78)
    in View (at AppEventHandlers.js:89)
    in AppEventHandlers (created by Connect(AppEventHandlers))
    in Connect(AppEventHandlers) (at ZulipMobile.js:22)
    in Provider (at StoreProvider.js:22)
    in StoreHydrator (at ZulipMobile.js:21)
    in CompatibilityChecker (at ZulipMobile.js:20)
    in Unknown (at renderApplication.js:35)
    in RCTView (at View.js:78)
    in View (at AppContainer.js:102)
    in RCTView (at View.js:78)
    in View (at AppContainer.js:122)
    in AppContainer (at renderApplication.js:34)
```

**Comment by pulkonet** [comment]
Will this be fixed on their end or we have to fix it on our side?

**Comment by jainkuniya** [comment]
as https://github.com/oblador/react-native-vector-icons/pull/693 got merged, it will be included in their next release, we just need to update when they release and close this :)

**Comment by borisyankov** [comment]
It was released a few days ago. Closing.

**Comment by gnprice** [comment]
This wasn't fixed. Here's a screenshot from master:

![image](https://user-images.githubusercontent.com/28173/41439637-6bcddfe8-6fe0-11e8-8255-e1a36aa60733.png)

It looks like the last release of the library was just before @jainkuniya 's PR was merged.

**Comment by ray-kraesig** [comment]
I've withdrawn my PR for now, but here's what was learned:

* The font-packaging issues seen in #2892 can now be resolved automatically via gradle (as seen in #3617, or as instructed in [the upstream docs](https://github.com/oblador/react-native-vector-icons#android)). iOS seems to be OK.
* Happily, despite [my original reason for withdrawal](https://github.com/zulip/zulip-mobile/pull/3617#issuecomment-532298318), no explicit tests for icon presence should be needed: the upstream package has added autogenerated Flow enums.
* Unhappily, the other Flow types upstream now provides seem to have a couple of mismatches, so we can't update to the latest version without fixing those.
  * At least one of those mismatches (the signature of `createIconSet`) is definitely an issue with upstream. The other may or may not be, and looks to require further investigation.

**Comment by gnprice** [comment]
Thanks for those details!

> At least one of those mismatches (the signature of `createIconSet`) is definitely an issue with upstream.

For things we determine are an issue with upstream and don't point to anything wrong in our code, I'd be ruthless in marking them `$FlowFixMe` -- with a ~1-line comment saying what seems to be wrong -- and moving on. Certainly if and when upstream fixes them that'll be welcome... but there are plenty of other things we can spend our time fixing instead.

For the others, we'll want to investigate far enough to either reach that same conclusion, or discover something wrong in our code and fix it :slightly_smiling_face:
